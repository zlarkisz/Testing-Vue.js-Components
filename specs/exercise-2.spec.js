import ExerciseForm from "@/exercise-2";
import { mount } from "@vue/test-utils";
import { nextTick } from "vue";

test("follow the user through the form", async () => {
  const wrapper = mount(ExerciseForm);

  expect(wrapper).toMatchSnapshot();

  const form = wrapper.find("form");
  const input = form.find("input");
  const tasksList = wrapper.find("ul");

  input.setValue("my todo");
  form.trigger("submit");

  await nextTick();

  expect(wrapper).toMatchSnapshot();
});
