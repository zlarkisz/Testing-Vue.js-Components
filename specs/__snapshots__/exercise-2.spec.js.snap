// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`follow the user through the form 1`] = `
<div>
  <form><input type="text" placeholder="Add a task"> <button type="submit">Add</button></form>
  <h2>Tasks</h2>
  <ul></ul>
</div>
`;

exports[`follow the user through the form 2`] = `
<div>
  <form><input type="text" placeholder="Add a task"> <button type="submit">Add</button></form>
  <h2>Tasks</h2>
  <ul>
    <li>
      my todo
      <button>Done</button>
    </li>
  </ul>
</div>
`;
